### PR TITLE
feat(android): Feature flag infra

### PR DESCRIPTION
**Pull request by ds300**
This PR resolves [CX-876]

### Description

This PR replaces the previous feature flag situation with a simpler setup. I'd advise to start by reading the new documentation to understand how things have changed https://github.com/artsy/eigen/blob/ds300/feature-flag-infra/docs/developing_a_feature.md

https://user-images.githubusercontent.com/1242537/106895982-d6601900-66e8-11eb-9cd8-f738c4668ff0.mp4

### Follow-up work

- Socialise the API change (emissionOption => featureFlag)
- Socialise the new feature flag definition and release process (link to docs, emphasise how much easier things are now). Advertise live demo with Q&A for next mobile practice meeting.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-876]: https://artsyproduct.atlassian.net/browse/CX-876

### CODE REVIEW

File: src/lib/utils/AdminMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+              Alert.alert(
+                "No Sentry DSN available",
+                __DEV__ ? "Set it in .env.shared and re-build the app." : undefined
+              )
+              return
+            }
+            throw Error("Sentry test error")
+          }}
+          chevron={null}
+        />
```
Comment (ds300) [inline src/lib/utils/AdminMenu.tsx:84]: https://user-images.githubusercontent.com/1242537/106896165-1de6a500-66e9-11eb-94c8-742681b55184.mp4

File: src/lib/utils/AdminMenu.tsx [hunk 2]
```diff
@@ -71,6 +71,13 @@ export const AdminMenu: React.FC<{ onClose(): void }> = ({ onClose = dismissModa
         <MenuItem
           title="Throw Sentry Error"
           onPress={() => {
+            if (!getCurrentEmissionState().sentryDSN) {
+              Alert.alert(
+                "No Sentry DSN available",
+                __DEV__ ? "Set it in .env.shared and re-build the app." : undefined
+              )
```
Comment (brainbicycle) [inline src/lib/utils/AdminMenu.tsx:78]: 🙏 nice

File: docs/developing_a_feature.md [hunk 3]
```diff
@@ -2,117 +2,94 @@
...
-<details><summary>What's wrong with logic or-ing lab options and Echo Features?</summary>
-
-The problem is that it conflates the responsibilities of lab options and Echo Features. Lab options are used for admins to see in-progress work; Echo Features are a safety valve so we can disable new features in the event of a catastrophic bug. If we ship a build that respects the Echo Feature but only has in-progress work, then users who install that version but don't upgrade to the fully-featured version will see that in-progress work.
-
-</details>
-
 ## Still Need Help?
 
-Ask for help in the #front-end-ios slack channel, we will be happy to assist!
+Ask for help in the #platform-mobile slack channel, we will be happy to assist!
```
Comment (brainbicycle) [inline docs/developing_a_feature.md:95]: Is this a channel should this be `#practice-mobile` ?

File: src/lib/utils/AdminMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+export const AdminMenu: React.FC<{ onClose(): void }> = ({ onClose = dismissModal }) => {
+  return (
+    <Flex
+      style={{
+        position: "absolute",
+        top: 0,
+        left: 0,
+        right: 0,
+        bottom: 0,
```
Comment (brainbicycle) [inline src/lib/utils/AdminMenu.tsx:25]: **question**: why do these need to be explicitly specified would have assumed defaulted to 0?

File: docs/developing_a_feature.md (hunk 3 above)
Comment (ds300) [reply docs/developing_a_feature.md:95]: Good catch!

```suggestion
Ask for help in the #practice-mobile slack channel, we will be happy to assist!
```

File: src/lib/utils/AdminMenu.tsx (hunk 4 above)
Comment (ds300) [reply src/lib/utils/AdminMenu.tsx:25]: 0 is not the default value. The default position/size depends on the node's position in the render tree as well as its children's dimensions. This overrides everything to make sure it always occupies the same space as it's parent.

Comment (brainbicycle) [reply src/lib/utils/AdminMenu.tsx:25]: Got it thanks for explaining!

File: src/lib/store/__tests__/ConfigModel-tests.ts [hunk 5]
```diff
@@ -2,28 +2,100 @@ import fetchMock from "jest-fetch-mock"
...
+    GlobalStore.actions.config.setEchoState({
+      ...echoLaunchJSON,
+      features: [
+        { name: "FeatureAEchoKey", value: false },
+        { name: "FeatureBEchoKey", value: true },
+      ],
+    })
+    expect(getComputedFeatures().FeatureB).toBe(false)
+  })
+})
```
Comment (brainbicycle) [inline src/lib/store/__tests__/ConfigModel-tests.ts:97]: Nice 🎉

File: Artsy/App/ARAppDelegate+Echo.m [hunk 6]
```diff
@@ -14,11 +14,7 @@ - (void)setupEcho
         [aero checkForUpdates:^(BOOL updatedDataOnServer) {
             if (!updatedDataOnServer) return;
             
-            [aero update:^(BOOL updated, NSError *error) {
-                if (!ARAppStatus.isRunningTests) {
-                    [[ARAppDelegate sharedInstance] updateEmissionOptions];
-                }
-            }];
+            [aero update:^(BOOL updated, NSError *error) {}];
```
Comment (brainbicycle) [inline Artsy/App/ARAppDelegate+Echo.m:17]: **nitpick:** if we are not using the block here believe we should just be able to pass nil:
```suggestion
            [aero update:nil];
```

**Review by brainbicycle — APPROVED** [review]
🏁 Awesome work, looks great! Thanks for the doc updates! I think this is a good one to share in mobile-practice to make more teams aware of changes.

File: Artsy/App/ARAppDelegate+Echo.m (hunk 6 above)
Comment (ds300) [reply Artsy/App/ARAppDelegate+Echo.m]: ah this is a type error because the parameter is _Nonnull, i'll revert the commit

**Pull request merged**